components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDAO.java

```diff
@@ -766,6 +817,105 @@ private void removeOauthOIDCPropertyTable(Connection connection,
         }
     }
 
+
+    /**
+     * Add scope validators for consumerApp using connection
+     *
+     * @param connection      same db connection used in OAuth creation
+     * @param appId           id of consumerApp
+     * @param scopeValidators list of scope validators.
+     * @throws SQLException sql error
+     */
+    private void addScopeValidators(Connection connection, int appId, String[] scopeValidators) throws SQLException {
+        if (scopeValidators != null && scopeValidators.length > 0) {
+            try (PreparedStatement stmt = connection.prepareStatement(SQLQueries.OAuthAppDAOSQLQueries
+                    .ADD_APP_SCOPE_VALIDATOR)) {
+                for (String scopeValidator : scopeValidators) {
+                    stmt.setInt(1, appId);
+                    stmt.setString(2, scopeValidator);
+                    stmt.addBatch();
+                }
+                stmt.executeBatch();
+            }
+        }
+    }
+
+    /**
+     * Retrieve all scope validators for specific appId
+     *
+     * @param connection same db connection used in retrieving OAuth App
+     * @param id         appId of the OAuth app
+     * @return list of scope validator names
+     * @throws SQLException sql error
+     */
+    private String[] getScopeValidators(Connection connection, int id) throws SQLException {
+        List<String> scopeValidators = new ArrayList<>();
+        try (PreparedStatement stmt = connection.prepareStatement(SQLQueries.OAuthAppDAOSQLQueries
+                .GET_APP_SCOPE_VALIDATORS)) {
+            stmt.setInt(1, id);
+            try (ResultSet rs = stmt.executeQuery()) {
+                while (rs.next()) {
+                    scopeValidators.add(rs.getString(1));
+                }
+            }
+        }
+        return scopeValidators.toArray(new String[scopeValidators.size()]);
```

Comment:
Could we please add some more debug logs to this class? E.g. Can add debug logs on how many scopevalidators found